src/main/java/io/vlingo/actors/InboundGridActorControl.java

```diff
@@ -114,4 +114,12 @@ public void relocate(Id receiver, Id sender, Definition.SerializationProxy defin
     });
     actor.resumeFromRelocation();
   }
+
+  @Override
+  public void recover(Id recipient, Id sender, Definition.SerializationProxy definitionProxy, Address address) {
+    logger.debug("Processing: Received application message: Recover");
+    final Optional<GridActor<?>> maybeActor = Optional.ofNullable((GridActor<?>) grid.actorAt(address));
+    final GridActor<?> actor = maybeActor.orElseGet(() -> (GridActor<?>) grid.actorAt(address));
+    actor.resumeFromRelocation(); // TODO Not enough for lattice models
```

Comment:
@VaughnVernon A solution would be to override `resumeFromRelocation` in lattice models to call `restore` and then the implementation in `super`. The implementation of a model's `id()` should be valid depending on what the application developer did in their model's constructor. WDYT?

Reply:
@buritos Seems reasonable. Perhaps we must insist on knowing the `id` of the entity by the time of `super()` construction. (I discussed this in Slack #lattice earlier.) However, recall that `start()` is sort of like a constructor but gives the extender the opportunity to do things to determine its id after `super(...)` is called. Sometimes `super()` gets in the way because you can't do anything prior.

Reply:
I meant `super.resumeFromRelocation` after doing `restore` in the `resumeFromRelocation ` override in the lattice-model abstract classes.

I have considered calling `start` here, and I guess that it would work for lattice-models, but I am not sure about the implications for non-lattice-model actors. WDYT?

Reply:
@buritos The `start()` is always sent as an async message when an actor is first constructed. It will be received some time later and asynchronously, but soon. All actors receive it and you can implement it in your base classes.

Reply:
In GridActors the start message is sent to the remote. Thunks, used for relaying messages to the active remote instance, do not process the start message. In a recover scenario, the thunk or a newly created thunk will have to resume from where the lost active instance left off.

Reply:
@buritos Huh. Shouldn't the `start()` be sent by the local hosting `Grid` that does the actual creation? In fact if you use `grid.actorFor(...)` in the local hosting `Grid`, `start()` should always be sent.

Reply:
Sorry, yes, the local sends start to the remote. I guess the best way to explain this is that GridMailbox always sends all messages to remote, except for stop which is handled both locally and sent to the remote.

Reply:
@buritos Which means that you can ask for the `id()` during `start()` rather than require it at super construction. This further reduces the total forced dependencies on the abstract base classes.

Reply:
@VaughnVernon I feel that this has gone a bit off-topic. I will soon push the changes for models and we can continue the model conversation there, then get back to this one.